src/RPC/RPC.php

```diff
@@ -43,6 +49,8 @@ public function __construct(RelayInterface $relay, CodecInterface $codec = null)
     {
         $this->relay = $relay;
         $this->codec = $codec ?? new JsonCodec();
+        /** @psalm-suppress DeprecatedProperty */
+        $this->hasSequence = \method_exists($this->relay, 'getNextSequence');
```

Comment:
What about a separated interface for this method?
```php
interface SequenceCarrierInterface extends RelayInterface {
    public function getNextSequenceNumber(): int;
}
```

or `SequencialRelayInterface`